File: trainning/uri/1 - Basics/1929/Main.cpp
#include <iostream>
#include <string>
#include <cstring>

using namespace std;
 
int main() {
 
    /**
     * Escreva a sua solução aqui
     * Code your solution here
     * Escriba su solución aquí
     */
     string linha;
     getline (cin,linha);

     char chr[linha.size()];
     linha.copy(chr, linha.size());

     int a = atoi(strtok(chr, " "));
     int b = atoi(strtok(NULL, " "));
     int c = atoi(strtok(NULL, " "));
     int d = atoi(strtok(NULL, " "));

     int tenta1 = (abs(b-c) < a && a < (b+c)) || (abs(a-c) < b && b < (a+c)) || (abs(a-b) < c && c < (a+b)); // a b c
     int tenta2 = (abs(c-d) < b && b < (c+d)) || (abs(b-d) < c && c < (b+d)) || (abs(b-c) < d && d < (b+c)); // b c d
     int tenta3 = (abs(c-d) < a && a < (c+d)) || (abs(a-d) < c && c < (a+d)) || (abs(a-c) < d && d < (a+c)); // a c d
     int tenta4 = (abs(b-d) < a && a < (b+d)) || (abs(a-d) < b && b < (a+d)) || (abs(a-b) < d && d < (a+b)); // a b d

     cout << ((tenta1 || tenta2 || tenta3 || tenta4) ? 'S' : 'N') << endl;

     return 0;
}

File: trainning/uri/1 - Basics/1929/Main.java
import java.io.IOException;
import java.util.Scanner;
 
/**
 * IMPORTANT: 
 *      O nome da classe deve ser "Main" para que a sua solução execute
 *      Class name must be "Main" for your solution to execute
 *      El nombre de la clase debe ser "Main" para que su solución ejecutar
 */
public class Main {
 
    public static void main(String[] args) throws IOException {
 
        /**
         * Escreva a sua solução aqui
         * Code your solution here
         * Escriba su solución aquí
         */
        Scanner scanner = new Scanner(System.in);
        String[] abcd = scanner.nextLine().split(" ");
        int a = Integer.parseInt(abcd[0]);
        int b = Integer.parseInt(abcd[1]);
        int c = Integer.parseInt(abcd[2]);
        int d = Integer.parseInt(abcd[3]);

        boolean tenta1 = (Math.abs(b-c) < a && a < (b+c)) || (Math.abs(a-c) < b && b < (a+c)) || (Math.abs(a-b) < c && c < (a+b)); // a b c
        boolean tenta2 = (Math.abs(c-d) < b && b < (c+d)) || (Math.abs(b-d) < c && c < (b+d)) || (Math.abs(b-c) < d && d < (b+c)); // b c d
        boolean tenta3 = (Math.abs(c-d) < a && a < (c+d)) || (Math.abs(a-d) < c && c < (a+d)) || (Math.abs(a-c) < d && d < (a+c)); // a c d
        boolean tenta4 = (Math.abs(b-d) < a && a < (b+d)) || (Math.abs(a-d) < b && b < (a+d)) || (Math.abs(a-b) < d && d < (a+b)); // a b d
        
        if (tenta1 || tenta2 || tenta3 || tenta4) { // vê se é triangulo
            System.out.println("S");
        }
        else {
            System.out.println("N");
        } 
    }
 
}
